applications/ConstitutiveLawsApplication/custom_constitutive/structural_elements_constitutive_laws/thickness_integrated_isotropic_constitutive_law.cpp

```diff
@@ -0,0 +1,766 @@
+    Vector& rValue
+    )
+{
+    return CalculateValue<Vector>(rParameterValues, rThisVariable, rValue);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Matrix& ThicknessIntegratedIsotropicConstitutiveLaw::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    return CalculateValue<Matrix>(rParameterValues, rThisVariable, rValue);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ThicknessIntegratedIsotropicConstitutiveLaw::InitializeMaterial(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues
+    )
+{
+    KRATOS_TRY
+
+    // Resizing first
+    mConstitutiveLaws.resize(mThicknessIntegrationPoints);
+
+    // We create the inner constitutive laws
+    const auto it_cl_begin = rMaterialProperties.GetSubProperties().begin();
+    auto& r_sub_prop = *(it_cl_begin);
+
+    for (IndexType i_layer = 0; i_layer < mConstitutiveLaws.size(); ++i_layer) {
+        KRATOS_ERROR_IF_NOT(r_sub_prop.Has(CONSTITUTIVE_LAW)) << "No constitutive law set" << std::endl;
+        mConstitutiveLaws[i_layer] = r_sub_prop[CONSTITUTIVE_LAW]->Clone();
+        mConstitutiveLaws[i_layer]->InitializeMaterial(r_sub_prop, rElementGeometry, rShapeFunctionsValues);
+    }
+
+    KRATOS_DEBUG_ERROR_IF(mConstitutiveLaws.size() == 0) << "ThicknessIntegratedIsotropicConstitutiveLaw: No CL defined" << std::endl;
+
+    KRATOS_CATCH("InitializeMaterial")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ThicknessIntegratedIsotropicConstitutiveLaw::CalculateCoordinatesAndWeights(
+    std::vector<double> &rCoordinates,
+    std::vector<double> &rWeights,
+    const IndexType NumberOfPoints,
+    const Properties& rMaterialProperties
+)
+{
+    KRATOS_TRY
+
+    rCoordinates.resize(NumberOfPoints);
+    rWeights.resize(NumberOfPoints);
+
+    const double thickness = rMaterialProperties[THICKNESS];
+    const double half_thickness = 0.5 * thickness;
+
+    // Composite Simpson rule weights
+    std::vector<double> simpson_weights(NumberOfPoints, 1.0);
+
+    if (NumberOfPoints >= 3) {
+        for (IndexType i = 1; i < NumberOfPoints - 1; ++i) {
+            simpson_weights[i] = (i % 2 == 0) ? 2.0 : 4.0;
+        }
+
+        // Normalize to sum = 1.0
+        double total_weight = std::accumulate(simpson_weights.begin(), simpson_weights.end(), 0.0);
+        for (auto &w : simpson_weights) {
+            w /= total_weight;
+        }
+    }
+
+    // Generate locations (from top to bottom: +t/2 -> -t/2)
```

Comment:
I may have missed it, but how about `SHELL_OFFSET`? I think the old Section class defines this one.